jsonargparse_tests/test_dataclass_like.py

```diff
@@ -628,3 +798,13 @@ def test_field_factory(self, parser):
         assert cfg1.data.p1 == ["one", "two"]
         assert cfg1.data.p1 == cfg2.data.p1
         assert cfg1.data.p1 is not cfg2.data.p1
+
+    def test_field_init_false(self, parser):
+        # This tests the following error:
+        #
+        # TypeError('Validation failed: Key "data.p2" is required but
+        # not included in config object or its value is None.')
+
+        parser.add_argument("--data", type=AttrsFieldInitFalse)
+        cfg = parser.parse_args(["--data.p1=1.0"])
+        assert cfg.data.p1 == 1.0
```

Comment:
Shouldn't this be testing data.p2. And what is the point of having p1 in the class and tested here?